packages/backend/src/managers/applicationManager.ts

```diff
@@ -202,25 +185,15 @@ export class ApplicationManager {
     });
   }
 
-  async restartContainerWhenModelServiceIsUp(
-    engineId: string,
-    modelServiceEndpoint: string,
-    container: ContainerAttachedInfo,
-  ): Promise<void> {
-    const alive = await isEndpointAlive(modelServiceEndpoint);
-    if (alive) {
-      const sampleAppContainerInspectInfo = await containerEngine.inspectContainer(engineId, container.name);
-      if (!sampleAppContainerInspectInfo.State.Running) {
-        await containerEngine.startContainer(engineId, container.name);
-      }
+  async waitContainerIsRunning(engineId: string, container: ContainerAttachedInfo): Promise<void> {
+    const sampleAppContainerInspectInfo = await containerEngine.inspectContainer(engineId, container.name);
+    if (sampleAppContainerInspectInfo.State.Running) {
       return;
     }
     await timeout(5000);
-    await this.restartContainerWhenModelServiceIsUp(engineId, modelServiceEndpoint, container).catch(
-      (error: unknown) => {
-        console.error('Error monitoring endpoint', error);
-      },
-    );
+    await this.waitContainerIsRunning(engineId, container).catch((error: unknown) => {
+      console.error('Error monitoring container', error);
+    });
   }
```

Comment:
If the app takes time to start, the recurse call it will create a high call stack. A for-loop would be more appropriate IMO